charts/k8ssandra/values.yaml

```
@@ -169,6 +169,66 @@ stargate:
     secret: ""
     username: ""
 
+  ingress:
+    # -- Optional hostname used to match requests
+    host:
+    auth:
+      enabled: true
```

Should this have a comment?

```suggestion
      # -- Enables authentication ingress
      enabled: true
```

---

Follow-up question, should we fail if authentication is disabled, but the other endpoints are enabled?

---

I considered that as well. Prior to this refactor, it was not possible to enable Stargate ingress without enabling it for auth. I had made that choice deliberately at the time, thinking that if `ingress.traefik.stargate.enabled=true` was sufficient. Some experience showed that layers of `enabled` are pretty obnoxious, and so I didn't want to keep that aspect of the structure. In the new setup, there is no `stargate.ingress.enabled` -- only `stargate.ingress.[auth|rest|graphql|cassandra].enabled`. They're all on by default (and, of course, they have no effect unless you've also installed an ingress controller).

It would be weird to disable auth but enable other aspects of Stargate ingress, but I don't think it would be invalid. Consider the possibility that auth could be handled by some kind of second- or third-party SSO service running in the same Kubernetes cluster, and having direct access to Stargate auth is actively undesirable.

---

I need to dig into the stargate auth service a bit more. If it's not required to talk with stargate then I don't care, but if it's required to use graphql or rest, maybe we tie whether it is enabled with something like `or stargate.ingress.rest.enabled stargate.ingress.graphql.enabled` instead of a dedicated field / toggle.

---

When we first introduced support for Stargate, authentication had to be enabled. It did not work when it was disabled. @jakerobb do you know if this has changed?

---

@jsanda this is just whether to enable ingress for Stargate's auth resource, not whether Stargate is configured to use or not use authentication.